joomla/joomla-cms
# [4.0] WebAssetManager: Speed up enable/disable asset

### Summary of Changes

WebAssetManager calculate dependencies on "request" (when it really need) instead of each call of `enableAsset`/`disableAsset`.
This will speed up operations like:
```php
$wa->enableAsset('foo')
  ->enableAsset('bar')
  ->disableAsset('bear')
  ->enableAsset('beer');
```

Additionally I have updated a comment for a Manager interface, added missed throws tags.


### Testing Instructions
Apply the patch and make sure everything still work :wink: 

ping @wilsonge 

for reference #22435
